server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/InMemoryMetricManager.java

```diff
@@ -0,0 +1,175 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.monitor;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import com.oceanbase.odc.service.monitor.MonitorAutoConfiguration.BusinessMeterRegistry;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Counter.Builder;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+@Getter
+@ConditionalOnProperty(value = "odc.system.monitor.actuator.enabled", havingValue = "true")
+public class InMemoryMetricManager implements MetricManager, InitializingBean {
+
+    private final Map<MeterKey, CounterWrapper> COUNTER_MAP = new ConcurrentHashMap<>();
+    private final Map<Pair<String, MeterKey>, TimerSampleHolder> TIMER_SAMPLE_MAP = new ConcurrentHashMap<>();
+    @Autowired
+    private BusinessMeterRegistry businessMeterRegistry;
+    @Autowired
+    private MonitorProperties monitorProperties;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        registerGauge(MeterKey.ofMeter(MeterName.METER_COUNTER_HOLDER_COUNT), this.COUNTER_MAP::size);
+        registerGauge(MeterKey.ofMeter(MeterName.METER_TIMER_HOLDER_COUNT), this.TIMER_SAMPLE_MAP::size);
+    }
+
+
+    public void registerGauge(MeterKey meterKey, Supplier<Number> f) {
+        MeterName meterName = meterKey.getMeterName();
+        Iterable<Tag> tags = meterKey.getTags();
+        Gauge.Builder<Supplier<Number>> builder = Gauge.builder(meterName.getMeterName(), f)
+                .description(MeterName.CONNECT_SESSION_ACTIVE_COUNT.getDescription());
+        for (Tag tag : tags) {
+            builder.tag(tag.getKey(), tag.getValue());
+        }
+        builder.register(businessMeterRegistry);
+    }
+
+    public void incrementCounter(MeterKey meterKey) {
+        CounterWrapper wrapper = COUNTER_MAP.get(meterKey);
+        if (wrapper == null) {
+            if (COUNTER_MAP.size() >= monitorProperties.getMeter().getMaxCounterMeterNumber()) {
```

Comment:
is this interface expected to be thread safe?
multi thread can entry wrapper == null logic to create same meter key